_proposals/pup-0001.rst

```diff
@@ -0,0 +1,207 @@
+
+* "Pulp Update Proposal" (PUP)
+* "Pulp Change Proposal" (PCP)
+* <Your idea here>
+
+
+Tracking Proposals
+==================
+
+Proposals would be in restructured text format and sent as a pull request against the
+pulpproject.org repo or a dedicated repo. We could easily create a tab at the top of pulpproject.org
+to show off merged proposals. This could also be in a separate, dedicated repo like Python does, but
+for simplicity I'm modeling this using the pulpproject.org repo. The idea to render proposals is
+similar to how [Python Enhancement Proposals (PEPs)](https://www.python.org/dev/peps/pep-0001/)
+work, which are markdown documents that are also viewable/readable as rendered documents via a
+browser.
+
+
+Finding/Discovering Proposals
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+An index mimicking [the Python one](https://www.python.org/dev/peps/) would be created which would
+show all proposals in different states (active, approved, rejected, abandoned). These would also be
+viewable via pulpproject.org in a dedicated area.
+
+
+Proposal Lifecycle
+==================
+
+Before Proposing
+^^^^^^^^^^^^^^^^
+
+Discussing an idea via the list prior to writing a formal proposal is encouraged. This is designed
+to save the author time by gathering feedback quickly before writing a concrete, detailed proposal.
+
+Announcement
+^^^^^^^^^^^^
+
+Once a proposal is merged to the proposals area, an announcement would be sent to pulp-dev, which
+would include a very brief introduction and a link to the proposal in its web viewable form.
+
+Discussion and Feedback
+^^^^^^^^^^^^^^^^^^^^^^^
+
+General discussion would occur on the announcement thread via pulp-dev. Threaded e-mail readers
+should make this conversation format easy to follow, while keeping the one proposal document in-tact
+and easy to read.
+
+
+Revision
+^^^^^^^^
+
+Revision would occur as additional pull requests against the active proposal. This allows the full
+history of a proposal to always be maintained. This is how Python does it. This also allows
+edits/updates to come with a commit message indicating the concerns addressed or improvements.
+Collaboration is also easy with a collaborator sending revisions to the author as a patch or a new
+PR against the proposal.
+
+
+Expressing Support/Concern
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The +1, +0, -0, -1 [feedback process of the Django community](https://docs.djangoproject.com/en/dev/internals/contributing/bugs-and-features/#how-we-make-decisions)
+would be used. These would be sent as responses via the pulp-dev mailing list. A -1 feedback must
+come with a statement of objective concerns explaining what harm will come from adopting the
+proposal. -1 feedback should not be used to express subjective preferences. Note that votes can come
+from anyone, not just core devs. This is directly modeled after the Django process [3] who modeled
+their process after the Python and Apache communities.
```

Comment:
How would the +0 and -0 work w/ this? I listed this as an alternative we should consider. I'd like to get more input from others. Maybe we should do this?